src/net/sourceforge/kolmafia/request/GenericRequest.java

```diff
@@ -1599,6 +1605,12 @@ private boolean sendRequest() {
       RequestLogger.printLine(MafiaState.ERROR, message);
       this.timeoutCount = TIMEOUT_LIMIT;
       return true;
+    } finally {
+      // At 10k client requests, the server will send a GOAWAY exception.
+      // We reset our connection before that to avoid this.
+      if (clientRequestsSent.incrementAndGet() >= 9900) {
```

Comment:
No magic number, please. Make a static constant - and put the comment there.

Reply:
Fixed in https://github.com/kolmafia/kolmafia/pull/750/commits/c7fd8de2136549feb5271298c3970b452d93445d